### PR TITLE
Fully manage aws-auth configmap file

### PR DESCRIPTION
**Pull request by max-rocket-internet**
Related issue: https://github.com/terraform-aws-modules/terraform-aws-eks/issues/50

The problem: Any users, roles or accounts that need access to the EKS cluster must have the relevant entries in the `aws-auth` configmap inside k8s. Currently this module only puts the ARN of the worker group in there, thus forcing people to manage this file manually, which is a pain.

Limitations:

1. It's a bit of a mess due to needing to loop over values in a template. Not sure if the new template syntax in [0.12](https://www.hashicorp.com/blog/terraform-0-1-2-preview) will help. Let me know if there's a better way to do this!

2. We can't have multiple groups per user or role due to this error: `lookup() may only be used with flat maps, this map contains elements of type list in`. No big deal I think at this stage.

3. We can't have interpolation in the items being passed to the module as it breaks the count option. Details [here](https://github.com/hashicorp/terraform/issues/10857). Annoying but also not a big deal.

Example configuration:

```hcl
module "cluster_1" {
  source                    = "git@github.com:terraform-aws-modules/terraform-aws-eks.git?ref=v1.4.0"
  cluster_name              = "cluster_1"
  subnets                   = ["${aws_subnet.public.*.id}"]
  vpc_id                    = "${aws_vpc.eks.id}"

  worker_groups = [
    {
      instance_type = "t2.medium"
      key_name      = "default"
    }
  ]

  map_roles = [
     {
       role_arn = "${aws_iam_role.k8s_users.arn}"
       username = "k8s_users"
       group = "system:masters"
     }
   ]

   map_users = [
     {
       user_arn = "arn:aws:iam::66666666666:user/user.name"
       username = "user.name"
       group  = "system:masters"
     }
   ]

   map_accounts = [
     "666666666666",
     "999999999999"
   ]
}
```

### CODE REVIEW

**Comment by max-rocket-internet** [comment]
1. Done
2. Done but I'm having to guess the date it would be merged and also the tag name. Is that correct? Wouldn't this be done after creating a tag?
3. Done. Does it look OK?

File: README.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ Generate them like so:
 
 ```bash
 go get github.com/segmentio/terraform-docs
-terraform-docs md ./ | cat -s | ghead -n -1 > README.md
```
Comment (brandoconnor) [inline README.md]: This line didn't include the `ghead` bit just for giggles. 😉 Let's keep it UNIXy 'round these parts and preserve those end of file newlines, eh?

Comment (brandoconnor) [reply README.md]: If that means moving from `ghead` to `head`, so be it. Maybe it's my own bias but I assume most people are developing on Macs which don't come with `ghead` but it's the closest thing to our Linux tool cousin.

Comment (brandoconnor) [reply README.md]: Actually, scrap the above comment. Let's do something that works for both modern Linux and Mac flavors. Mind changing it to the following:

```bash
terraform-docs md ./ | cat -s | tail -r | tail -n +2 | tail -r > README.md
```

**Comment by brandoconnor** [comment]
Can you figure out what's going on with the failed `terraform fmt` and fix up my comment around the `terraform-docs` command? We're sooooo close!

**Comment by brandoconnor** [comment]
This is great btw. Thanks again!

File: README.md (hunk 1 above)
Comment (brandoconnor) [reply README.md]: It's a utility that does the same thing as unix `head`. The problem is you have to install it via brew. `tail` should be shipped standard on any mac unless I'm off my rocker.

**Comment by max-rocket-internet** [comment]
I'm still resolving conflicts.

**Comment by max-rocket-internet** [comment]
OK I was a little confused after all my other PRs were merged and then I had `exec: "aws-iam-authenticator": executable file not found in $PATH` but now resolved.

1. terraform fmt done
2. docs change tested and done

Anything else?

**Pull request merged**